langchain/document_transformers/interrogate.py

```diff
@@ -0,0 +1,32 @@
+"""Transform documents"""
+from typing import Any, Sequence
+
+from pydantic import BaseModel
+
+from langchain.utils import get_from_dict_or_env
+from langchain.schema import BaseDocumentTransformer, Document
+from doctran import Doctran
+
+
+class DocumentInterrogator(BaseDocumentTransformer, BaseModel):
```

Comment:
We might consider making this a `BaseBlobParser` similar to what we do [here](https://github.com/hwchase17/langchain/blob/705d2f5b9225e63ad4603776d5047d18a6c3fdfa/langchain/document_loaders/parsers/grobid.py#L14C20-L14C34).

And then it is called easily w/o any boilerplate loader code:

```
loader = GenericLoader.from_filesystem(
    "./example_data/pds",
    glob="*",
    suffixes=[".pdf"],
    parser=DocumentTranslator()
)
docs = loader.load()
```

Reply:
Sounds good I'll take a look at adding that. These transformers depend on an OpenAI call so need to be run async - would that cause any issues when used as a parser?

Reply:
No, we have a similar parser that uses OpenAI Whisper.